### ISSUE TITLE
quarkus image build is unable to build a blank quarkus app

### ISSUE THREAD

**Issue by hu1buerger** [quarkusio/quarkus#44672]
### Describe the bug

`quarkus image build` not able to build a clean quarkus app.

### Expected behavior

I would expect that `quarkus image build` can build a docker image for a newly create quarkus app (`quarkus create app explorviz.span-service --gradle`)

### Actual behavior

```sh
quarkus image build
To honour the JVM settings for this build a single-use Daemon process will be forked. For more on this, please refer to https://docs.gradle.org/8.9/userguide/gradle_daemon.html#sec:disabling_the_daemon in the Gradle documentation.
Daemon will be stopped at the end of the build 

FAILURE: Build failed with an exception.

* What went wrong:
Problem configuring task :imageBuild from command line.
> Unknown command-line option '--builder'.

* Try:
> Run gradlew help --task :imageBuild to get task usage details.
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
> Get more help at https://help.gradle.org.

BUILD FAILED in 30s
````

### How to Reproduce?

Steps to reproduce this:
1. docker run --rm  -v /var/run/docker.sock:/var/run/docker.sock -it ghcr.io/graalvm/graalvm-community:21 bash
2. curl -Ls https://sh.jbang.dev | bash -s - app setup && source /root/.bashrc
3. jbang app install --fresh --force quarkus@quarkusio
4. quarkus create app explorviz.span-service --gradle
5. cd explorviz.span-service/
6. quarkus ext add container-image-docker # is this necessary?
7. quarkus image build

### Output of `java -version`

21

### Quarkus version or git rev

3.16.4

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

8.9

**Comment by geoand** [comment]
@iocanel IIUC, doing `quarkus image build` isn't yet supported for Gradle?

**Comment by iocanel** [comment]
> [@iocanel](https://github.com/iocanel) IIUC, doing `quarkus image build` isn't yet supported for Gradle?

It should work for gradle too, even when the container-image-docker is not added to the project.

**Comment by geoand** [comment]
Yeah, I could also reproducer, which is why I asked about the state of this :).

Do you plan on tackling it?

Thanks

**Comment by Hu1buerger** [comment]
@geoand i would volunteer to write a regression test for this case and maybe some others related, if that is something the quarkus project would be interested in.

**Comment by geoand** [comment]
Let's see what @iocanel has to say

**Comment by Malte-Hansen** [comment]
On my local machine, `quarkus image build` works with Quarkus version `3.15.1`  (and Java 21) but fails with Quarkus version `3.16.1` for a project created with `quarkus create app explorviz.span-service --gradle`

**Comment by geoand** [comment]
Is this still an issue with the latest Quarkus versions?

**Comment by geoand** [comment]
Closing for lack of feedback
